src/frontend/hooks/server/observations.ts

```diff
@@ -36,13 +37,14 @@ export function useCreateObservation() {
   const project = useProject();
 
   return useMutation({
-    mutationFn: async ({
-      value,
-    }: {
-      value: Parameters<MapeoProject['observation']['create']>[0];
-    }) => {
+    mutationFn: async ({value}: {value: ClientGeneratedObservation}) => {
       if (!project) throw new Error('Project instance does not exist');
-      return project.observation.create(value);
+
+      return project.observation.create({
+        ...value,
+        schemaName: 'observation',
+        attachments: [],
+      });
```

Comment:
According to the typings, this should be a `promise` but it is being returned as `{"_h": 0, "_i": 0, "_j": null, "_k": null, "clear": [Function anonymous]}`

This is causing a problem with our loading and success state, as nothing is being resolved or rejected.

It is still creating an observation though.

Reply:
I looked into this. I checked the tests on rpc-reflector to make sure this is being tested, and I tried adding this code to CoMapeo (in the project context):

```js
project.observation.create({ ... }).then(console.log)
```

This worked successfully, resolving and logging the result. What you are seeing when you log the value being returned is because of the way that rpc-reflector uses a `Proxy`, and how Hermes implements it. This is what I see on mobile:

```js
const p = project.observation.create({ ... })
console.log(p) // {"_h": 0, "_i": 0, "_j": null, "_k": null, "clear": [Function anonymous]}
console.log('then' in p) // true
console.log(typeof p.then) // function
```

TL;DR As far as I can test it, `project.observation.create()` is returning a promise, and it does resolve, and it also rejects if an invalid value is passed. I think the error must be elsewhere.

Reply:
Update: I also tried running this PR as-is on an Android emulator. I created a new observation and pressed save, and both `onSuccess` handlers correctly fired (I checked by adding console.log), and it navigated back to the map screen. Not sure what behaviour is not working, let me know and I can investigate further.